common/src/indexer_service/http/request_handler.rs

```diff
@@ -41,16 +44,40 @@ where
         .with_label_values(&[&manifest_id.to_string()])
         .inc();
 
+    #[derive(Debug, serde::Deserialize)]
+    pub struct QueryBody {
+        pub query: String,
+        pub variables: Option<Box<RawValue>>,
+    }
+
     let request =
         serde_json::from_slice(&body).map_err(|e| IndexerServiceError::InvalidRequest(e.into()))?;
 
     let mut attestation_signer: Option<AttestationSigner> = None;
 
     if let Some(receipt) = receipt.into_signed_receipt() {
         let allocation_id = receipt.message.allocation_id;
+        let signature = receipt.signature;
```

Comment:
I don't understand how/why you're using the signature

Reply:
Signature is the best "unique id" for a specific receipt. It is unique for a given receipt + signer.

Reply:
I thought you wanted it to be unique per "SQL text" vs. unique per each query.